app/models/mixins/has_virtualization_manager_mixin.rb

```diff
@@ -0,0 +1,29 @@
+module HasVirtualizationManagerMixin
+  extend ActiveSupport::Concern
+
+  def virtualization_endpoint_created(role)
+    if role == "kubevirt" && virtualization_manager.nil?
+      virtualization_manager = ensure_virtualization_manager
+      virtualization_manager.save
+    end
+  end
+
+  def virtualization_endpoint_destroyed(role)
+    if role == "kubevirt" && virtualization_manager.present?
+      virtualization_manager.destroy_queue
+    end
+  end
+
+  private
+
+  def ensure_virtualization_manager
+    if virtualization_manager.nil?
+      build_virtualization_manager(:parent_manager  => self,
```

Comment:
Where is this (`build_virtualization_manager`) defined?  I don't see it in any of the three PRs

Reply:
@agrare it's rails method, the name was chosen to reflect the nature of that sub-provider, in the same sense as monitoring_manager was chosen for https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/has_monitoring_manager_mixin.rb#L22

The purpose of 'virtualization_manager' is to reflect the ability to serve virtual machines by a container provider, by that secondary provider.

If this doesn't make sense to you, I'll modify it.

Reply:
Right but in that case there actually is a manager type called MonitoringManager (https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/monitoring_manager.rb).  This should match the type of manager that is being added.